File: feature-dashboard/src/main/java/fr/o80/sample/featuredashboard/dagger/DashboardModule.java
package fr.o80.sample.featuredashboard.dagger;

import dagger.Module;
import dagger.Provides;
import fr.o80.sample.featuredashboard.DashboardService;
import fr.o80.sample.lib.core.LibConfiguration;

/**
 * @author Olivier Perez
 */
@Module
public class DashboardModule {

    @Provides
    @FeatureScope
    DashboardService provideDashboardService(LibConfiguration configuration) {
        return new DashboardService(configuration);
    }

}

File: feature-dashboard/src/main/java/fr/o80/sample/featuredashboard/dagger/DashboardModule.kt
package fr.o80.sample.featuredashboard.dagger

import dagger.Module
import dagger.Provides
import fr.o80.sample.featuredashboard.DashboardService
import fr.o80.sample.lib.core.LibConfiguration
import fr.o80.sample.lib.dagger.FeatureScope

/**
 * @author Olivier Perez
 */
@Module
class DashboardModule {

    @Provides
    @FeatureScope
    internal fun provideDashboardService(configuration: LibConfiguration): DashboardService {
        return DashboardService(configuration)
    }

}
